ctf/setup.py

```diff
@@ -49,3 +54,8 @@ def build_challenges():
                     db.session.commit()
                 except IntegrityError:
                     db.session.rollback()
+                    challenge = Challenge.query.filter_by(
+                            title=problem['title'])
+                    problem_dict.update({'id': challenge.first().id})
```

Comment:
Then perhaps if `build_problem_options` listed explicit dict key names? That would feel better than running a `copy()` and `del`ing stuff.

Something like this?

```python
problem = {
    'title': problem_config['title'],
    'fleg_hash': hash_fleg(problem_config['fleg']),
    ...
}
```